#! /usr/bin/env python

PACKAGE='aricc_2d_vision'
try:
    import imp
    imp.find_module(PACKAGE)
    from dynamic_reconfigure.parameter_generator_catkin import *
except:
    import roslib; roslib.load_manifest(PACKAGE)
    from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("noise_size", double_t, 0, "the size of noise contour", 1, 1, 1000)
gen.add("min_size", double_t, 0, "the min limit size of contour", 100, 1, 1000)
gen.add("min_dist", double_t, 0, "the min limit distance of merged contour", 20, 1, 1000)
gen.add("epsilon", double_t, 0, "the epsilon for poly detection", 3, 1, 1000)
gen.add("debug",    bool_t, 0, "the switch of debug", False)
gen.add("draw_contours",    bool_t, 0, "the switch of debug", False)
gen.add("draw_convex",    bool_t, 0, "the switch of debug", False)
gen.add("draw_poly",    bool_t, 0, "the switch of debug", False)
gen.add("close",    bool_t, 0, "the switch of close", True)

exit(gen.generate(PACKAGE, "aricc_2d_vision", "ContourFinder"))
